extension/observer/kafkatopicsobserver/extension_test.go

```diff
@@ -0,0 +1,128 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package kafkatopicsobserver
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/IBM/sarama"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer/kafkatopicsobserver/internal/metadata"
+)
+
+type MockClusterAdmin struct {
+	sarama.ClusterAdmin
+	mock.Mock
+}
+
+func (m *MockClusterAdmin) ListTopics() (map[string]sarama.TopicDetail, error) {
+	args := m.Called()
+	return args.Get(0).(map[string]sarama.TopicDetail), args.Error(1)
+}
+
+func (m *MockClusterAdmin) Close() error {
+	return m.Called().Error(0)
+}
+
+func TestCollectEndpointsDefaultConfig(t *testing.T) {
+	factory := NewFactory()
+	mockAdmin := &MockClusterAdmin{}
+	// Override the createKafkaClusterAdmin function to return the mock admin
+	originalCreateKafkaClusterAdmin := createKafkaClusterAdmin
+	createKafkaClusterAdmin = func(_ context.Context, _ Config) (sarama.ClusterAdmin, error) {
+		return mockAdmin, nil
+	}
+
+	ext, err := newObserver(zap.NewNop(), factory.CreateDefaultConfig().(*Config))
+	require.NoError(t, err)
+	require.NotNil(t, ext)
+
+	obvs, ok := ext.(*kafkaTopicsObserver)
+	require.True(t, ok)
+	kEndpoints := obvs.ListEndpoints()
+
+	want := []observer.Endpoint{}
+	createKafkaClusterAdmin = originalCreateKafkaClusterAdmin
+	require.Equal(t, want, kEndpoints)
+}
+
+func TestCollectEndpointsAllConfigSettings(t *testing.T) {
```

Comment:
This test is leading to CI issues, see https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/38192

Reply:
Ok, I can see what causes the problem, I'll create a PR with this test fixed

Reply:
My PR with a fix #38218